### ISSUE TITLE
Oculus + shaders causing a constant leaking air effect from PneumaticCraft’s pressure chamber

### ISSUE THREAD

**Issue by Nevrai** [Asek3/Oculus#173]
### What happened?

The pressure chamber from [PneumaticCraft: Repressurized](https://www.curseforge.com/minecraft/mc-mods/pneumaticcraft-repressurized) starts emitting a dense fog from within it, filling the room and any nearby rooms with a thick white fog, after it hits a certain point of pressure—exactly 0.50/5 bar.

What’s happening is that the default effect of a few steam particles showing up only within the chamber itself instead appear as a dense white fog that fills up a huge radius. Disabling shaders solves this issue, so it’s definitely a problem with Oculus/shaders.

![2022-08-30_09 12 53](https://user-images.githubusercontent.com/59761487/187375421-9e0ad7c0-d0d4-4374-bf2b-14a1e1ea9f18.png)

Tested with [Complementary Shaders v4](https://www.curseforge.com/minecraft/customization/complementary-shaders) and [Complementary Reimagined](https://www.curseforge.com/minecraft/customization/complementary-reimagined), both having the same issue. Also tried turning off and turning down every fog-related effect in the shaders settings, but this had no effect on this bug.

### Versions
* **Modpack:** [All the Mods 7](https://www.curseforge.com/minecraft/modpacks/all-the-mods-7) (0.4.29—latest)
* **Minecraft:** 1.18.2
* **Forge:** 40.1.69
* **Oculus:** 1.2.5a (latest for 1.18.2)
* **Rubidium:** 0.5.3a (latest for 1.18.2)
* **PneumaticCraft: Repressurized:** 3.4.0-124 (latest for 1.18.2)
* **Java:** 17 (OpenJDK Platform Library, version 17.0.2+8)
* **OS:** Windows 11 Home, version 21H2
* **GPU:** NVIDIA GeForce RTX 3070 Laptop GPU

**Comment by Nevrai** [comment]
Huh, back in my world, this is still happening even when I deactivate the shaders. This is very strange. It may not be a shaders issue, but I suppose it could still be an Oculus issue, as it’s still installed. It’s just odd that this issue stopped happening when I turned off shaders in the test world with only the aforementioned mods installed and then started happening again with shaders toggled on again, but in my world in All the Mods 7 (+ some mods added such as Rubidium and Oculus), this happens regardless of whether shaders are enabled.

![2022-08-30_11 24 17](https://user-images.githubusercontent.com/59761487/187402127-dc5f919b-d6f1-4e7b-9b58-4d409c6a137d.png)

I’m not exactly sure what’s going on, but I suppose this might still be an Oculus issue.

**Comment by falxie** [comment]
Have you tried completely removing Oculus and not just disabling shaders? I've noticed that Oculus has effects even if shaders aren't enabled.

**Comment by IliasCrunch** [comment]
There is a ''fix'' that i tried out... It worked. Try to disable the particles in the pressure chamber in the config, should do the trick.

**Comment by Nevrai** [comment]
@IliasCrunch Yeah, I know about that. That is a recently added feature, which was added after I submitted this issue. In fact, I’m the one who suggested it, and the main PneumaticCraft: Repressurized dev, @desht, was kind enough to add this option just one day after I suggested it. It’s a decent workaround for now. Not seeing the particles is a shame, but at least it prevents this frustrating issue until Oculus fixes this, if it is fixed, and it’s far better not to see the particles within the chamber than to have PC:RP practically be unplayable because of this Oculus issue.

**Comment by desht** [comment]
I have a feeling it may also depend upon the specific shader in use as well.  I'm sure I've tested it with Oculus and not seen the problem, so I'll need to go back and check which shader I was using at the time.

**Comment by desht** [comment]
Update: yep, running the BSL v8.1.03 shader pack along with Rubidium 0.5.3a and Oculus 1.2.5a and particles are rendering fine. So it looks like a problem with certain specific shaders.
![2022-10-20_08 49 19](https://user-images.githubusercontent.com/736831/196889059-e204fad5-aa83-4721-9d75-50b3cd7e4f31.png)

**Comment by Vectrobe** [comment]
This bug has popped back up again after updating oculus to 1.6.13, and disabling shaders.
The problem only exists when shaders are disabled, so there is something Oculus is doing wrong when running in vanilla mode, probably leaving a GL setting uncleared.

**Comment by Vectrobe** [comment]
ok nevermind, disabling oculus also causes this to occur, so as it turns out its a bug specific to pneum's particle rendering, that turns out to get fixed again specifically when shaders are enabled.
